CSharp/DevVmPowershell/DevVmPsModules/Cmdlets/WaitForSmokeTestToFinishModule.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using System.Threading.Tasks;
+using Helpers.Implementations;
+using Helpers.Interfaces;
+
+namespace DevVmPsModules.Cmdlets
+{
+	[Cmdlet(VerbsCommon.Watch, "SmokeTestForTestToFinish")]
+	public class WaitForSmokeTestToFinishModule : BaseModule
+	{
+		[Parameter(
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
+			ValueFromPipeline = true,
+			Position = 0,
+			HelpMessage = "Name of the Relativity Instance")]
+		public string RelativityInstanceName { get; set; }
+
+		[Parameter(
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
+			ValueFromPipeline = true,
+			Position = 1,
+			HelpMessage = "Username of the Relativity Admin")]
+		public string RelativityAdminUserName { get; set; }
+
+		[Parameter(
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
+			ValueFromPipeline = true,
+			Position = 2,
+			HelpMessage = "Password of the Relativity Admin")]
+		public string RelativityAdminPassword { get; set; }
+
+		[Parameter(
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
+			ValueFromPipeline = true,
+			Position = 3,
+			HelpMessage = "Username of the Relativity Sql Account")]
+		public string SqlAdminUserName { get; set; }
+
+		[Parameter(
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
+			ValueFromPipeline = true,
+			Position = 4,
+			HelpMessage = "Password of the Relativity Sql Account")]
+		public string SqlAdminPassword { get; set; }
+
+		[Parameter(
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
+			ValueFromPipeline = true,
+			Position = 5,
+			HelpMessage = "Name of the Workspace that the Smoke Tests are being run in")]
+		public string WorkspaceName { get; set; }
+
+		protected override void ProcessRecordCode()
```

Comment:
Looks like the timeout value is hard coded in the Constants class. In future, we might want to update it easily but don't necessarily want to recompile the PS module code. Can we have the timeout value (in minutes) as one of the input to this PS module. That way, we could control it from the attributes/default.rb file in Chef code easily.